src/Entity.ts

```diff
@@ -33,18 +33,13 @@ export class Entity {
     return this._componentMap[classRef.name] as T;
   }
 
-  // TODO: Prevent users from being able to pass in
-  // a component subclass here. Should we instead
-  // only accept subclasses and return a new component
-  // construction? Only problem is that this will
-  // require every component to have an empty constructor,
-  // but I think that's something I'm ok with enforcing.
-  // Components should be dumb structs, and nothing else.
-  // If we go that route, we should also start returning
-  // the new component here.
```

Comment:
Oof, just saw this. And it'd make some things a lot nicer. Should probably leave in an `addComponent` field that has this behavior? I can see it being a common usecase